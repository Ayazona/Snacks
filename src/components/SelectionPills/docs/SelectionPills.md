
### Self controlled state

`SelectionPills` maintains its own internal for the array of pills provided. The
initial selected state of a pill may be passed as a property of the pill object,
but all changes will be maintained by `SelectionPills` component.

The array of `pills` all for all the properties of a `SelectionPill` to be passed
to override the underlying pill behavior or style.

```jsx
// import { SelectionPills } from 'ic-snacks'
import SelectionPills from '../SelectionPills.js'

const initialState = { activePill: 'carrots' }
const pills = [
  { text: 'bananas', id: 'filterby-1', isSelected: true },
  { text: 'carrots', id: 'filterby-2' },
  { text: 'watermelon', id: 'filterby-3' },
  { text: 'snacks', id: 'filterby-4' },
  { text: 'kale', id: 'filterby-5' },
  { text: 'endives', id: 'filterby-6', isDisabled: true },
  { text: 'arugula', id: 'filterby-7' },
  { text: 'spinach', id: 'filterby-8' },
  { text: 'potatoes', id: 'filterby-9' },
]

;<div>
  <SelectionPills
    pills={pills}
    onSelectPill={(e, pill) => {
      e.preventDefault()
      console.log('SelectionPills SelectionPill clicked!', pill)
    }}
    label="Filter by"
  />
  </div>
```

#### With select all pill

Optional parameter to include select all button that unselects all selected options
when clicked.

```jsx
// import { SelectionPills } from 'ic-snacks'
import SelectionPills from '../SelectionPills.js'

const pills = [
  { text: 'bananas', id: 'selection-1', isSelected: true },
  { text: 'carrots', id: 'selection-2' },
  { text: 'watermelon', id: 'selection-3' },
  { text: 'snacks', id: 'selection-4' },
  { text: 'kale', id: 'selection-5' },
  { text: 'endives', id: 'selection-6', isDisabled: true },
  { text: 'arugula', id: 'selection-7' },
  { text: 'spinach', id: 'selection-8' },
  { text: 'potatoes', id: 'selection-9' },
]
;<div>
  <SelectionPills
    includeSelectAll
    selectAllLabel="Select all"
    pills={pills}
    onSelectPill={(e, pill) => {
      e.preventDefault()
      console.log('SelectionPills SelectionPill clicked!', pill)
    }}
    label="Filter by"
  />
  </div>
```

#### With maximum number of selections

Optional parameter to restrict the number of selections that can be chosen

```jsx
// import { SelectionPills } from 'ic-snacks'
import SelectionPills from '../SelectionPills.js'

const pills = [
  { text: 'bananas', id: 'maxSelection-1' },
  { text: 'carrots', id: 'maxSelection-2' },
  { text: 'watermelon', id: 'maxSelection-3' },
  { text: 'snacks', id: 'maxSelection-4' },
  { text: 'kale', id: 'maxSelection-5' },
  { text: 'endives', id: 'maxSelection-6' },
  { text: 'arugula', id: 'maxSelection-7' },
  { text: 'spinach', id: 'maxSelection-8' },
  { text: 'potatoes', id: 'maxSelection-9' },
]
;<div>
  <SelectionPills
    maxSelectionCount={3}
    pills={pills}
    onSelectPill={(e, pill) => {
      e.preventDefault()
      console.log('SelectionPills SelectionPill clicked!', pill)
    }}
    label="Select up to 3:"
  />
  </div>
```

### Parent controlled state

State of each `SelectionPill` can be controlled by a parent component by passing a
in the `parentControlledState` prop. When this prop is present, the `pills`
array prop controls the state of each pill through the `isSelected` property.



```jsx
// import { SelectionPills } from 'ic-snacks'
import SelectionPills from '../SelectionPills.js'

const initialState = {
  pills: [
    { text: 'bananas', id: 'parentControlled-1' },
    { text: 'carrots', id: 'parentControlled-2' },
    { text: 'watermelon', id: 'parentControlled-3' },
    { text: 'snacks', id: 'parentControlled-4' },
    { text: 'kale', id: 'parentControlled-5' },
    { text: 'endives', id: 'parentControlled-6' },
    { text: 'arugula', id: 'parentControlled-7' },
    { text: 'spinach', id: 'parentControlled-8' },
    { text: 'potatoes', id: 'parentControlled-9' },
  ]
}
;<div>
  <SelectionPills
    parentControlledState
    pills={state.pills}
    onSelectPill={(e, pill) => {
      e.preventDefault()
      console.log('SelectionPills SelectionPill clicked!', pill)
      const updatedPills = state.pills.map(p => {
        if (p.id === pill.id) {
          return {
            ...p,
            isSelected: !p.isSelected
          }
        }
        return p
      })
      setState({
        pills: updatedPills
      })
    }}
    label="Filter by"
  />
  </div>
```
