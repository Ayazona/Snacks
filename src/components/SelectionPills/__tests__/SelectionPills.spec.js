import React from 'react'
import renderer from 'react-test-renderer'
import { StyleRoot } from 'radium'
import { mount } from 'enzyme'
import { spy } from 'sinon'
import SelectionPills from '../SelectionPills'

describe('SelectionPills', () => {
  const testPills = [
    {
      text: 'bananas',
      id: 'selection-1',
    },
    {
      text: 'arugula',
      id: 'selection-2',
    },
  ]

  it('renders SelectionPills correctly', () => {
    const tree = renderer
      .create(
        <StyleRoot>
          <SelectionPills
            pills={testPills}
            onSelectPill={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
          />
        </StyleRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders SelectionPills with select all included correctly', () => {
    const tree = renderer
      .create(
        <StyleRoot>
          <SelectionPills
            includeSelectAll
            selectAllLabel="Select all"
            pills={testPills}
            onSelectPill={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
          />
        </StyleRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders SelectionPills with pill attributes set correctly', () => {
    const attributesPills = [
      { text: 'bananas', id: 'selection-1', isSelected: true },
      { text: 'arugula', id: 'selection-2', isDisabled: true },
      { text: 'apples', id: 'selection-3', style: { button: { marginLeft: '10px' } } },
    ]
    const tree = renderer
      .create(
        <StyleRoot>
          <SelectionPills
            pills={attributesPills}
            onSelectPill={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
          />
        </StyleRoot>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('handles onPillClick correctly', () => {
    const onSelectPill = spy()
    const Pills = mount(
      <StyleRoot>
        <SelectionPills pills={testPills} onSelectPill={onSelectPill} />
      </StyleRoot>
    )

    Pills.find('input')
      .first()
      .simulate('change')
    expect(onSelectPill.called).toBeTruthy()
    expect(onSelectPill.withArgs(3)).toBeTruthy()
  })

  it('disables remaining pills when max selected', () => {
    const selectPills = [
      { text: 'bananas', id: 'selection-1' },
      { text: 'arugula', id: 'selection-2', isSelected: true },
      { text: 'apple', id: 'selection-3' },
    ]
    const Pills = mount(
      <StyleRoot>
        <SelectionPills pills={selectPills} maxSelectionCount={2} />
      </StyleRoot>
    )

    Pills.find('input')
      .first()
      .simulate('change')
    const applePill = Pills.find('input')
      .last()
      .instance()
    expect(applePill.disabled).toBe(true)
  })

  it('unselects other pills when select all is included', () => {
    const selectPills = [
      { text: 'bananas', id: 'selection-1', isSelected: true },
      { text: 'apple', id: 'selection-2' },
    ]
    const Pills = mount(
      <StyleRoot>
        <SelectionPills pills={selectPills} includeSelectAll />
      </StyleRoot>
    )

    const allPills = Pills.find('input')
    const bananasPill = allPills.at(1).instance()
    const applePill = allPills.at(2).instance()

    expect(bananasPill.checked).toBe(true)
    expect(applePill.checked).toBe(false)

    allPills.first().simulate('change')

    expect(bananasPill.checked).toBe(false)
    expect(applePill.checked).toBe(false)
  })

  xit('responds to parent controlled state correctly', () => {
    const selectPills = [
      { text: 'bananas', id: 'selection-1', isSelected: true },
      { text: 'apple', id: 'selection-2' },
    ]
    const Pills = mount(
      <StyleRoot>
        <SelectionPills pills={selectPills} parentControlledState />
      </StyleRoot>
    )
    const allPills = Pills.find('input')
    const bananasPill = allPills.first()
    expect(bananasPill.instance().checked).toBe(true)
    bananasPill.simulate('change')
    expect(bananasPill.instance().checked).toBe(true)

    const updatedPills = selectPills.map(p => {
      return {
        ...p,
        isSelected: false,
      }
    })
    console.log('UPDATED', updatedPills)
    Pills.setProps({ pills: updatedPills })
    expect(bananasPill.instance().checked).toBe(false)
  })
})
