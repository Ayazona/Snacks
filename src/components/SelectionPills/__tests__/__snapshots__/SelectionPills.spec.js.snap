// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionPills renders SelectionPills correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    onKeyDown={[Function]}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <button
      aria-label="back"
      className="rmq-4534e6b"
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#43B02A",
          "border": "0",
          "borderRadius": "50%",
          "boxShadow": "0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)",
          "color": "#FFFFFF",
          "display": "none",
          "height": "48px",
          "left": 8,
          "lineHeight": "1",
          "position": "absolute",
          "textAlign": "center",
          "top": "5px",
          "transition": "background-color 150ms ease-in-out",
          "width": "48px",
          "zIndex": 100,
        }
      }
    >
      <i
        aria-hidden={true}
        data-radium={true}
        style={
          Object {
            "fontFamily": "ic-icons",
            "fontSize": "20px",
            "fontSmoothing": "antialiased",
            "fontStyle": "normal",
            "fontVariant": "normal",
            "fontWeight": "normal",
            "lineHeight": "1",
            "osxFontSmoothing": "grayscale",
            "position": "relative",
            "speak": "none",
            "textTransform": "none",
          }
        }
      >
        
      </i>
    </button>
    <div
      className="rmq-16b35912"
      data-radium={true}
      style={
        Object {
          "position": "relative",
          "transform": "translate3d(0px, 0, 0)",
          "transition": "transform 150ms ease-in-out",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        data-radium={true}
        style={Object {}}
      >
        <div
          data-radium={true}
          style={
            Object {
              "display": "inline-block",
              "minWidth": "100%",
              "padding": 0,
            }
          }
        >
          <span
            id="selection_pills_label_Filter_by"
          >
            Filter by
          </span>
          <ul
            aria-labelledby="selection_pills_label_Filter_by"
            data-radium={true}
            style={
              Object {
                "display": "inline-block",
                "marginBottom": 16,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 16,
                "padding": 0,
              }
            }
          >
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={false}
                  disabled={false}
                  id="selection-1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-1"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                bananas
              </label>
            </li>
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={false}
                  disabled={false}
                  id="selection-2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-2"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                arugula
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <button
      aria-label="next"
      className="rmq-4534e6b"
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#43B02A",
          "border": "0",
          "borderRadius": "50%",
          "boxShadow": "0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)",
          "color": "#FFFFFF",
          "display": "none",
          "height": "48px",
          "lineHeight": "1",
          "position": "absolute",
          "right": 8,
          "textAlign": "center",
          "top": "5px",
          "transition": "background-color 150ms ease-in-out",
          "width": "48px",
          "zIndex": 100,
        }
      }
    >
      <i
        aria-hidden={true}
        data-radium={true}
        style={
          Object {
            "fontFamily": "ic-icons",
            "fontSize": "20px",
            "fontSmoothing": "antialiased",
            "fontStyle": "normal",
            "fontVariant": "normal",
            "fontWeight": "normal",
            "lineHeight": "1",
            "osxFontSmoothing": "grayscale",
            "position": "relative",
            "speak": "none",
            "textTransform": "none",
          }
        }
      >
        
      </i>
    </button>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-16b35912{left: inherit !important;
overflow-x: scroll !important;
display: block !important;
-ms-overflow-style: none !important;
overflow: -moz-scrollbars-none !important;
webkit-overflow-scrolling: touch !important;}}
@media (max-width: 767px){ .rmq-4534e6b{display: none !important;}}",
      }
    }
  />
</div>
`;

exports[`SelectionPills renders SelectionPills with pill attributes set correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    onKeyDown={[Function]}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <button
      aria-label="back"
      className="rmq-4534e6b"
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#43B02A",
          "border": "0",
          "borderRadius": "50%",
          "boxShadow": "0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)",
          "color": "#FFFFFF",
          "display": "none",
          "height": "48px",
          "left": 8,
          "lineHeight": "1",
          "position": "absolute",
          "textAlign": "center",
          "top": "5px",
          "transition": "background-color 150ms ease-in-out",
          "width": "48px",
          "zIndex": 100,
        }
      }
    >
      <i
        aria-hidden={true}
        data-radium={true}
        style={
          Object {
            "fontFamily": "ic-icons",
            "fontSize": "20px",
            "fontSmoothing": "antialiased",
            "fontStyle": "normal",
            "fontVariant": "normal",
            "fontWeight": "normal",
            "lineHeight": "1",
            "osxFontSmoothing": "grayscale",
            "position": "relative",
            "speak": "none",
            "textTransform": "none",
          }
        }
      >
        
      </i>
    </button>
    <div
      className="rmq-16b35912"
      data-radium={true}
      style={
        Object {
          "position": "relative",
          "transform": "translate3d(0px, 0, 0)",
          "transition": "transform 150ms ease-in-out",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        data-radium={true}
        style={Object {}}
      >
        <div
          data-radium={true}
          style={
            Object {
              "display": "inline-block",
              "minWidth": "100%",
              "padding": 0,
            }
          }
        >
          <span
            id="selection_pills_label_Filter_by"
          >
            Filter by
          </span>
          <ul
            aria-labelledby="selection_pills_label_Filter_by"
            data-radium={true}
            style={
              Object {
                "display": "inline-block",
                "marginBottom": 16,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 16,
                "padding": 0,
              }
            }
          >
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={true}
                  disabled={false}
                  id="selection-1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-1"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "rgba(67, 176, 42, 0.1)",
                    "borderColor": "#43B02A",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                bananas
              </label>
            </li>
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={false}
                  disabled={true}
                  id="selection-2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-2"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "#F7F7F7",
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "rgba(67, 176, 42, 0.6)",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                arugula
              </label>
            </li>
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={false}
                  disabled={false}
                  id="selection-3"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-3"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "10px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                apples
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <button
      aria-label="next"
      className="rmq-4534e6b"
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#43B02A",
          "border": "0",
          "borderRadius": "50%",
          "boxShadow": "0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)",
          "color": "#FFFFFF",
          "display": "none",
          "height": "48px",
          "lineHeight": "1",
          "position": "absolute",
          "right": 8,
          "textAlign": "center",
          "top": "5px",
          "transition": "background-color 150ms ease-in-out",
          "width": "48px",
          "zIndex": 100,
        }
      }
    >
      <i
        aria-hidden={true}
        data-radium={true}
        style={
          Object {
            "fontFamily": "ic-icons",
            "fontSize": "20px",
            "fontSmoothing": "antialiased",
            "fontStyle": "normal",
            "fontVariant": "normal",
            "fontWeight": "normal",
            "lineHeight": "1",
            "osxFontSmoothing": "grayscale",
            "position": "relative",
            "speak": "none",
            "textTransform": "none",
          }
        }
      >
        
      </i>
    </button>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-16b35912{left: inherit !important;
overflow-x: scroll !important;
display: block !important;
-ms-overflow-style: none !important;
overflow: -moz-scrollbars-none !important;
webkit-overflow-scrolling: touch !important;}}
@media (max-width: 767px){ .rmq-4534e6b{display: none !important;}}",
      }
    }
  />
</div>
`;

exports[`SelectionPills renders SelectionPills with select all included correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    onKeyDown={[Function]}
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <button
      aria-label="back"
      className="rmq-4534e6b"
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#43B02A",
          "border": "0",
          "borderRadius": "50%",
          "boxShadow": "0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)",
          "color": "#FFFFFF",
          "display": "none",
          "height": "48px",
          "left": 8,
          "lineHeight": "1",
          "position": "absolute",
          "textAlign": "center",
          "top": "5px",
          "transition": "background-color 150ms ease-in-out",
          "width": "48px",
          "zIndex": 100,
        }
      }
    >
      <i
        aria-hidden={true}
        data-radium={true}
        style={
          Object {
            "fontFamily": "ic-icons",
            "fontSize": "20px",
            "fontSmoothing": "antialiased",
            "fontStyle": "normal",
            "fontVariant": "normal",
            "fontWeight": "normal",
            "lineHeight": "1",
            "osxFontSmoothing": "grayscale",
            "position": "relative",
            "speak": "none",
            "textTransform": "none",
          }
        }
      >
        
      </i>
    </button>
    <div
      className="rmq-16b35912"
      data-radium={true}
      style={
        Object {
          "position": "relative",
          "transform": "translate3d(0px, 0, 0)",
          "transition": "transform 150ms ease-in-out",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        data-radium={true}
        style={Object {}}
      >
        <div
          data-radium={true}
          style={
            Object {
              "display": "inline-block",
              "minWidth": "100%",
              "padding": 0,
            }
          }
        >
          <span
            id="selection_pills_label_Filter_by"
          >
            Filter by
          </span>
          <ul
            aria-labelledby="selection_pills_label_Filter_by"
            data-radium={true}
            style={
              Object {
                "display": "inline-block",
                "marginBottom": 16,
                "marginLeft": 4,
                "marginRight": 4,
                "marginTop": 16,
                "padding": 0,
              }
            }
          >
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  aria-label="Select all"
                  checked={false}
                  id="selectAllPill"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selectAllPill"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                Select all
              </label>
            </li>
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={false}
                  disabled={false}
                  id="selection-1"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-1"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                bananas
              </label>
            </li>
            <li
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <div
                data-radium={true}
                style={
                  Object {
                    "border": 0,
                    "clip": "rect(0 0 0 0)",
                    "clippath": "inset(50%)",
                    "height": 1,
                    "margin": -1,
                    "overflow": "hidden",
                    "padding": 0,
                    "position": "absolute",
                    "whiteSpace": "nowrap",
                    "width": 1,
                  }
                }
              >
                <input
                  checked={false}
                  disabled={false}
                  id="selection-2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  type="checkbox"
                />
              </div>
              <label
                data-radium={true}
                htmlFor="selection-2"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "borderColor": "#E0E0E0",
                    "borderRadius": 24,
                    "borderStyle": "solid",
                    "borderWidth": 1,
                    "color": "#43B02A",
                    "display": "block",
                    "fontSize": 14,
                    "fontWeight": 600,
                    "height": 32,
                    "lineHeight": "32px",
                    "marginBottom": 0,
                    "marginLeft": "4px",
                    "marginRight": "4px",
                    "marginTop": 0,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "transition": "background-color 150ms ease-in-out",
                  }
                }
              >
                arugula
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <button
      aria-label="next"
      className="rmq-4534e6b"
      data-radium={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#43B02A",
          "border": "0",
          "borderRadius": "50%",
          "boxShadow": "0 1px 2px 0 rgba(0, 0, 0, 0.26), 0 1px 4px 0 rgba(0, 0, 0, 0.16)",
          "color": "#FFFFFF",
          "display": "none",
          "height": "48px",
          "lineHeight": "1",
          "position": "absolute",
          "right": 8,
          "textAlign": "center",
          "top": "5px",
          "transition": "background-color 150ms ease-in-out",
          "width": "48px",
          "zIndex": 100,
        }
      }
    >
      <i
        aria-hidden={true}
        data-radium={true}
        style={
          Object {
            "fontFamily": "ic-icons",
            "fontSize": "20px",
            "fontSmoothing": "antialiased",
            "fontStyle": "normal",
            "fontVariant": "normal",
            "fontWeight": "normal",
            "lineHeight": "1",
            "osxFontSmoothing": "grayscale",
            "position": "relative",
            "speak": "none",
            "textTransform": "none",
          }
        }
      >
        
      </i>
    </button>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-16b35912{left: inherit !important;
overflow-x: scroll !important;
display: block !important;
-ms-overflow-style: none !important;
overflow: -moz-scrollbars-none !important;
webkit-overflow-scrolling: touch !important;}}
@media (max-width: 767px){ .rmq-4534e6b{display: none !important;}}",
      }
    }
  />
</div>
`;
